

<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>Benchmarks &mdash; Medaka 0.3.0 documentation</title>
  

  
  
  
  

  

  
  
    

  

  <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="Walkthrough" href="walkthrough.html" />
    <link rel="prev" title="Installation" href="installation.html" /> 

  
  <script src="_static/js/modernizr.min.js"></script>

</head>

<body class="wy-body-for-nav">

   
  <div class="wy-grid-for-nav">

    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search">
          

          
            <a href="index.html" class="icon icon-home"> Medaka
          

          
          </a>

          
            
            
              <div class="version">
                0.3.0
              </div>
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <ul class="current">
<li class="toctree-l1"><a class="reference internal" href="installation.html">Installation</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">Benchmarks</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#comparison-of-medaka-and-nanopolish">Comparison of <cite>medaka</cite> and <cite>nanopolish</cite></a></li>
<li class="toctree-l2"><a class="reference internal" href="#evaluation-across-samples-and-depths">Evaluation across samples and depths</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="walkthrough.html">Walkthrough</a></li>
<li class="toctree-l1"><a class="reference internal" href="development.html">Development</a></li>
<li class="toctree-l1"><a class="reference internal" href="future.html">History and Future Directions</a></li>
</ul>
<ul>
<li class="toctree-l1"><a class="reference internal" href="medaka.html">medaka package</a></li>
</ul>

            
          
        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="index.html">Medaka</a>
        
      </nav>


      <div class="wy-nav-content">
        
        <div class="rst-content">
        
          















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="index.html">Docs</a> &raquo;</li>
        
      <li>Benchmarks</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
            
            <a href="_sources/benchmarks.rst.txt" rel="nofollow"> View page source</a>
          
        
      </li>
    
  </ul>

  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <div class="section" id="benchmarks">
<h1>Benchmarks<a class="headerlink" href="#benchmarks" title="Permalink to this headline">¶</a></h1>
<p>The following demonstrates the utility of Medaka’s neural network in forming an
improved consensus from a pileup of reads.</p>
<p>Results were obtained using the default model provided with <cite>medaka</cite>. This model
was trained using data obtained from E.coli, S.cerevisaie and H.sapiens samples.
Training data were basecalled using Guppy 0.3.0. Draft assemblies were created
using the <a class="reference external" href="https://nanoporetech.github.io/pomoxis/examples.html#fast-de-novo-assembly">mini_assemble</a>
pipeline in <a class="reference external" href="https://github.com/nanoporetech/pomoxis">pomoxis</a>.</p>
<p>Error statistics were calculated using the <a class="reference external" href="https://github.com/nanoporetech/pomoxis">pomoxis</a> program <cite>stats_from_bam</cite> after
aligning 100kb chunks of the consensus to the reference. Reported metrics are
median values over all chunks.</p>
<div class="section" id="comparison-of-medaka-and-nanopolish">
<h2>Comparison of <cite>medaka</cite> and <cite>nanopolish</cite><a class="headerlink" href="#comparison-of-medaka-and-nanopolish" title="Permalink to this headline">¶</a></h2>
<p>Evaluation of the model was performed using the <cite>medaka</cite> E.coli
<a class="reference internal" href="walkthrough.html"><span class="doc">Walkthrough</span></a> dataset. These data we not used to train the model.
Basecalling was performed with <a class="reference external" href="https://github.com/nanoporetech/scrappie">scrappie v1.3.2</a> using the <cite>rgrgr_r94</cite> model. The
pileup had a median depth of ~80-fold. <a class="reference external" href="https://github.com/jts/nanopolish">nanopolish v0.10.1</a> was run with homopolymer correction but
without methylation correction. <cite>medaka</cite> and <cite>nanopolish</cite> were run on the same
hardware.</p>
<table border="1" class="docutils">
<colgroup>
<col width="46%" />
<col width="22%" />
<col width="32%" />
</colgroup>
<thead valign="bottom">
<tr class="row-odd"><th class="head">&#160;</th>
<th class="head">medaka</th>
<th class="head">nanopolish</th>
</tr>
</thead>
<tbody valign="top">
<tr class="row-even"><td>Q(Accuracy)</td>
<td>31.55</td>
<td>31.22</td>
</tr>
<tr class="row-odd"><td>Q(Identity)</td>
<td>46.02</td>
<td>45.23</td>
</tr>
<tr class="row-even"><td>Q(Deletion)</td>
<td>32.60</td>
<td>31.81</td>
</tr>
<tr class="row-odd"><td>Q(Insertion)</td>
<td>39.21</td>
<td>43.01</td>
</tr>
<tr class="row-even"><td>runtime (hours)</td>
<td>0.27</td>
<td>1.05</td>
</tr>
<tr class="row-odd"><td>CPU cores</td>
<td>4</td>
<td>48</td>
</tr>
<tr class="row-even"><td>CPU hours</td>
<td>1.06</td>
<td>50.4</td>
</tr>
</tbody>
</table>
<p>For this dataset <cite>medaka</cite> delivers similar results to <cite>nanopolish</cite> in a fraction
of the time.</p>
</div>
<div class="section" id="evaluation-across-samples-and-depths">
<h2>Evaluation across samples and depths<a class="headerlink" href="#evaluation-across-samples-and-depths" title="Permalink to this headline">¶</a></h2>
<p>Evaluation of the model was performed using E.coli, H.sapiens chromosome 21, and
<a class="reference external" href="https://github.com/rrwick/Basecalling-comparison">K.pneumoniae</a>.  The E.coli
and human reads were basecalled with <cite>Guppy</cite> version 0.3.0, while the Klebsiella
reads were basecalled with <a class="reference external" href="https://github.com/nanoporetech/scrappie">scrappie v1.3.2</a> using the <cite>rgrgr_r94</cite> model. The
draft assemblies here were created at multiple depths using the <a class="reference external" href="https://nanoporetech.github.io/pomoxis/examples.html#fast-de-novo-assembly">mini_assemble</a>
pipeline in <a class="reference external" href="https://github.com/nanoporetech/pomoxis">pomoxis</a>.</p>
<table border="1" class="docutils">
<colgroup>
<col width="32%" />
<col width="20%" />
<col width="21%" />
<col width="26%" />
</colgroup>
<thead valign="bottom">
<tr class="row-odd"><th class="head">Data set</th>
<th class="head">Racon Error (%)</th>
<th class="head">Medaka Error (%)</th>
<th class="head">Nanopolish Error (%)</th>
</tr>
</thead>
<tbody valign="top">
<tr class="row-even"><td>E.coli 25X</td>
<td>0.291</td>
<td>0.125</td>
<td>0.127</td>
</tr>
<tr class="row-odd"><td>E.coli 50X</td>
<td>0.247</td>
<td>0.079</td>
<td>0.080</td>
</tr>
<tr class="row-even"><td>E.coli  75X</td>
<td>0.238</td>
<td>0.065</td>
<td>0.069</td>
</tr>
<tr class="row-odd"><td>E.coli 100X</td>
<td>0.228</td>
<td>0.063</td>
<td>0.060</td>
</tr>
<tr class="row-even"><td>E.coli 150X</td>
<td>0.231</td>
<td>0.065</td>
<td>0.057</td>
</tr>
<tr class="row-odd"><td>E.coli 200X</td>
<td>0.231</td>
<td>0.061</td>
<td>0.052</td>
</tr>
<tr class="row-even"><td>L.brevis 250X</td>
<td>0.293</td>
<td>0.055</td>
<td>0.047</td>
</tr>
<tr class="row-odd"><td>K.pneumoniae <a class="footnote-reference" href="#id6" id="id5">[1]</a> 200X</td>
<td>0.576</td>
<td>0.318</td>
<td>0.086</td>
</tr>
</tbody>
</table>
<table class="docutils footnote" frame="void" id="id6" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label"><a class="fn-backref" href="#id5">[1]</a></td><td>native (non-PCRed) data. Nanopolish was run with the –methylation-aware=dcm,dam
option.</td></tr>
</tbody>
</table>
<p><cite>medaka</cite> produces similar results to Nanopolish (on PCRd data) in a fraction of
the time, and provides a marked benefit over Racon of native DNA.</p>
</div>
</div>


           </div>
           
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="walkthrough.html" class="btn btn-neutral float-right" title="Walkthrough" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right"></span></a>
      
      
        <a href="installation.html" class="btn btn-neutral" title="Installation" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <p>
        &copy; Copyright 2017, Oxford Nanopore Technologies.

    </p>
  </div>
  Built with <a href="http://sphinx-doc.org/">Sphinx</a> using a <a href="https://github.com/rtfd/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>

        </div>
      </div>

    </section>

  </div>
  


  

    <script type="text/javascript">
        var DOCUMENTATION_OPTIONS = {
            URL_ROOT:'./',
            VERSION:'0.3.0',
            LANGUAGE:'None',
            COLLAPSE_INDEX:false,
            FILE_SUFFIX:'.html',
            HAS_SOURCE:  true,
            SOURCELINK_SUFFIX: '.txt'
        };
    </script>
      <script type="text/javascript" src="_static/jquery.js"></script>
      <script type="text/javascript" src="_static/underscore.js"></script>
      <script type="text/javascript" src="_static/doctools.js"></script>
      <script type="text/javascript" src="https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>

  

  <script type="text/javascript" src="_static/js/theme.js"></script>

  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>